pxtlib/emitter/driver.ts

```diff
@@ -83,6 +83,26 @@ namespace ts.pxt {
         usedSymbols?: U.Map<SymbolInfo>; // q-names of symbols used
     }
 
+    export function computeUsedParts(resp: CompileResult): string[] {
+        if (!resp.usedSymbols)
+            return null;
+
+        let parts: string[] = [];
+        for (let symbol in resp.usedSymbols) {
+            let info = resp.usedSymbols[symbol]
+            if (info && info.attributes && (<any>info.attributes)["parts"]) {
+                let partsStr = <string>(<any>info.attributes)["parts"];
+                if (partsStr) {
+                    partsStr
+                        .split(" ")
+                        .filter(p => 0 < p.length && parts.indexOf(p) < 0)
```

Comment:
This will still add the part twice if it's "led led"; better just put an `if` statement in the `forEach`